deploy/cloudformation/config.go

```diff
@@ -34,10 +34,15 @@ type config struct {
 	EnrollmentToken       string     `mapstructure:"ENROLLMENT_TOKEN"`
 	ElasticArtifactServer *string    `mapstructure:"ELASTIC_ARTIFACT_SERVER"`
 	ElasticAgentVersion   string     `mapstructure:"ELASTIC_AGENT_VERSION"`
-	IntegrationType       *string    `mapstructure:"INTEGRATION"`
 	Dev                   *devConfig `mapstructure:"DEV"`
+	DeploymentType        string     `mapstructure:"DEPLOYMENT_TYPE"`
```

Comment:
I think it should be inside the `devConfig` because all the "regular" parameters being sent as CloudFormation parameters which will cause an unknown parameter.
Another option is to wrap the CloudFormation parameters under some new object.

Reply:
`Dev` is also not a part of the template and yet it exists in the same scope.
I do think that the `DeploymentType` should be in the root because it is not part of the `Dev` subsection.
It actually decides what template we will apply eventually.
This is only relevant for use-case when we are running via the `gomain.go`.